packages/algob/test/builtin-tasks/sign-multisig.ts

```diff
@@ -0,0 +1,129 @@
+import { compareArray } from "@algo-builder/runtime/src/lib/compare";
+import { decodeAddress, decodeSignedTransaction } from "algosdk";
+import { assert } from "chai";
+import * as fs from "fs";
+import path from "path";
+import sinon from 'sinon';
+
+import { TASK_SIGN_MULTISIG } from "../../src/builtin-tasks/task-names";
+import { ASSETS_DIR } from "../../src/internal/core/project-structure";
+import { loadSignedTxnFromFile } from "../../src/lib/files";
+import { HttpNetworkConfig } from "../../src/types";
+import { getEnv } from "../helpers/environment";
+import { useFixtureProject } from "../helpers/project";
+import { account1, bobAcc } from "../mocks/account";
+
+export const netCfg: HttpNetworkConfig = {
+  accounts: [account1, bobAcc],
+  host: "localhost",
+  port: 8080,
+  token: "some-fake-token"
+};
+
+describe("Sign-Multisig task", () => {
+  useFixtureProject("config-project");
+
+  const inputFile = 'multisig-signed.txn'; // present in config-project/assets
+  const outFile = 'bob-signed.txn';
+  const bobPk = decodeAddress(bobAcc.addr).publicKey;
+  before(async function () {
+    this.env = await getEnv(netCfg); // update this.env with custom netCfg
+  });
+
+  afterEach(function () {
+    const outPath = path.join(ASSETS_DIR, outFile);
+    if (fs.existsSync(outPath)) { fs.rmSync(outPath); }
+  });
+
+  it("Append bob's signature to multisigned txn loaded from file", async function () {
+    const encodedTx = loadSignedTxnFromFile(inputFile) as Uint8Array;
+    const txMsig = decodeSignedTransaction(encodedTx).msig;
+
+    // input assertions (verfiy bob pk is present in pre-image, but it is not signed,
+    // i.e signature is not present)
+    assert.isDefined(txMsig);
+    assert.deepInclude(txMsig?.subsig, { pk: bobPk } as any); // msig includes bobPk (but without signature i.e "s" field)
```

Comment:
do we need this part ? can we remove it ? or  move to another section.

Reply:
yes, we need it. We're trying to append bob signature to multisig, so we ensure first it's not present when loading the multisig from file. i.e `pk` is present but `s` isn't.